lib/services/provided/autocomplete.js

```diff
@@ -12,7 +14,7 @@ type CompleteReply = {
     _jupyter_types_experimental?: Array<{
       start?: number,
       end?: number,
-      text?: string,
+      text: string,
```

Comment:
Are you sure that the object will always have a text object (do you have the docs?)?

Reply:
There is no valid and general docs for this experimental completion AFAIK.
Only IPython kernel doc says like https://ipython.readthedocs.io/en/stable/api/generated/IPython.core.completer.html#IPython.core.completer.Completion

I made `text` non nullable since I changed `autocompleteSuggesion` type definition and it makes flow complains otherwise, but at the same time I can't think completion without completion text itself.

Reply:
And the case is still handled in the condition check in `parseCompletion` , thus I think it's enough.

Reply:
making text optional only means that it can be undefined. It can never be null. However as long as there are no bugs in use I am fine with the change